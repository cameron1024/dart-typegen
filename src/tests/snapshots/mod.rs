use crate::{codegen, model::Library};

#[test]
fn test_foo() {
    let kdl = include_str!("./foo.kdl");

    let config = Library::parse_impl(None, &kdl).unwrap();

    let mut generated = Vec::new();
    codegen::codegen(&mut generated, &config).unwrap();
    let generated = String::from_utf8(generated).unwrap();

    insta::assert_snapshot!(generated);
}
