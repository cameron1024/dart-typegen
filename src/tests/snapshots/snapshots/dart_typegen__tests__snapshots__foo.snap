---
source: src/tests/snapshots/mod.rs
expression: generated
---
final class Foo with EquatableMixin {
  final String name;
  final int age;

  const Foo({required this.name, required this.age});
}
